x-pack/plugins/security_solution/scripts/endpoint/common/fleet_services.ts

```diff
@@ -209,9 +215,70 @@ export const waitForHostToEnroll = async (
     );
   }
 
-  log.debug(`Host [${hostname}] has been enrolled with fleet`);
+  log.info(`Host [${hostname}] has been enrolled with fleet`);
   log.verbose(found);
 
+  if (!awaitMetadata) {
+    return found;
+  }
+  log.info('Querying indices');
+
+  log.debug(
+    JSON.stringify(
+      await esClient?.search({
+        index: [
+          '.fleet-agents',
+          'metrics-endpoint.metadata_current_default',
+          '.metrics-endpoint.metadata_united_default',
+        ],
+      }),
+      null,
+      2
+    )
+  );
+  log.info('Successfully queried indices.');
+
+  log.info(`Awaiting host to show up in Metadata`);
+
+  const metadataLookupStartedTime = new Date();
+  const hasTimedOutMetadataLookup = (): boolean => {
+    const elapsedTime = Date.now() - metadataLookupStartedTime.getTime();
+    return elapsedTime > timeoutMs;
+  };
+  let metadataFound: HostInfo | undefined;
+
+  while (!metadataFound && !hasTimedOutMetadataLookup()) {
+    metadataFound = await retryOnError(
+      async () =>
+        fetchEndpointMetadataList(kbnClient, {
+          kuery: `united.endpoint.agent.id:  "${agentId}" or united.endpoint.host.hostname: "${hostname}"`,
+        }).then((response) => {
+          return response.data.filter(
+            (record) =>
+              record.metadata.host.hostname === hostname && record.host_status === 'healthy'
+          )[0];
+        }),
+      RETRYABLE_TRANSIENT_ERRORS
+    );
+
+    if (!metadataFound) {
+      await new Promise((r) => setTimeout(r, 2000));
+    }
+  }
+
+  if (!metadataFound) {
+    throw Object.assign(
+      new Error(
+        `Timed out waiting for host [${hostname}] to show up in Metadata. Waited ${
+          timeoutMs / 1000
+        } seconds`
+      ),
+      { agentId, hostname }
+    );
+  }
+
+  log.info(`Host [${hostname}] is healthy.`);
```

Comment:
yeah, all of this seems like duplicate code with `waitForEndpointToStreamData()`